### PR TITLE
Features/137005157/multi vendor fix

### PR DESCRIPTION
**Pull request by victor-enogwe**
### What does this PR do?
The PR adds the multi-vendor feature that allows a user to own a shop and add products to the shop. The shop owner can also view orders for his shop

### Description of Task to be completed?
When a user registers, he/she can choose to register as a shop owner (vendor) and create a shop.
A user can also upgrade his/her account to a vendor account

### What are the relevant pivotal tracker stories?
#137005157 Marketplace / Multi-Vendor

###Screenshots (if appropriate)
oie_7z7ewijfcfvy

### CODE REVIEW

File: private/data/Products.json [hunk 1]
```diff
@@ -31,7 +31,13 @@
   "updatedAt": {
     "$date": "2014-06-01T12:17:13.949-0700"
   },
-  "vendor": "Example Manufacturer"
+  "vendor": "Reaction",
+  "vendorDetail" : {
+        "userId" : "ZSDt9Euyo7FeGNjYw",
+        "shopName" : "telle",
+        "shopPhone" : "07066900500",
+        "shopAddress" : "how do you do today"
```
Comment (andela-aolaniran) [inline private/data/Products.json]: Can we use an address here ?

File: build/config.gypi [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# Do not edit. File was generated by node-gyp's "configure" step
```
Comment (vonvick) [inline build/config.gypi]: what does this configurations do?

File: client/modules/accounts/templates/profile/profile.js [hunk 3]
```diff
@@ -64,5 +101,97 @@ Template.accountProfile.helpers({
...
+      errors.shopName = {
+        i18nKeyReason: "Hello! Shop name has already been taken.",
+        reason: "Hello! Shop name has already been taken."
+      };
+    }
+    if (!/\d+(-)?/g.test(shopPhone) && shopPhone.length <= 14) {
+      errors.shopPhone = { i18nKeyReason: "invalid shop phone number", reason: "invalid shop phone number"};
+    }
+
+    if (!/\w+/g.test(shopAddress) && shopAddress.length <= 250) {
```
Comment (vonvick) [inline client/modules/accounts/templates/profile/profile.js:143]: could you explain why you are checking if the ShopAddress'  is less than 250 characters. What happens if the address is valid but less than 250 characters

File: client/modules/accounts/templates/signUp/signUp.js [hunk 4]
```diff
@@ -53,6 +66,36 @@ Template.loginFormSignUpView.events({
...
+      shopPhone = template.$(".shop-phone").val().trim();
+      shopAddress = template.$(".shop-address").val().trim();
+
+      if (!/\w+/g.test(shopName) && shopName.length <= 20) {
+        errors.shopName = { i18nKeyReason: "invalid shop name", reason: "invalid shop name"};
+      }
+
+      if (allShops.includes(shopName)) {
+        errors.shopName = {
+          i18nKeyReason: "shop name already exists. please choose another.",
```
Comment (vonvick) [inline client/modules/accounts/templates/signUp/signUp.js:81]: Why don't we use the shopId instead

File: lib/collections/schemas/accounts.js [hunk 5]
```diff
@@ -115,3 +115,18 @@ export const Accounts = new SimpleSchema({
     optional: true
   }
 });
+
+
+/**
+ * Reaction Schemas Vendors
+ */
+export const Vendors = new SimpleSchema({
+  theId: {
```
Comment (vonvick) [inline lib/collections/schemas/accounts.js:129]: A good property name would be nice

File: private/data/Products.json (hunk 1 above)
Comment (vonvick) [reply private/data/Products.json]: a valid address would be nice too

**Review by vonvick — COMMENTED** [review]
Looks good to me except for the changes that were pointed out.

File: build/config.gypi (hunk 2 above)
Comment (victor-enogwe) [reply build/config.gypi]: an auto generated build config, most likely because of selenium testing

File: private/data/Products.json (hunk 1 above)
Comment (victor-enogwe) [reply private/data/Products.json]: Noted

File: client/modules/accounts/templates/profile/profile.js (hunk 3 above)
Comment (victor-enogwe) [reply client/modules/accounts/templates/profile/profile.js:143]: We are disallowing addresses greater than 250 characters.

File: imports/plugins/core/ui/client/containers/mediaGalleryContainer.js [hunk 6]
```diff
@@ -160,7 +164,18 @@ function composer(props, onData) {
   if (viewAs === "customer") {
     editable = false;
   } else {
-    editable = Reaction.hasPermission(props.permission || ["createProduct"]);
+    let vendor = null;
```
[the remark this answers is not in the document]
Comment (olawalejarvis) [reply imports/plugins/core/ui/client/containers/mediaGalleryContainer.js]: Fixed

File: client/modules/accounts/templates/profile/profile.html [hunk 7]
```diff
@@ -1,5 +1,14 @@
 <template name="accountProfile">
-
+  <style>
```
Comment (andela-jare) [inline client/modules/accounts/templates/profile/profile.html:2]: Why not create a css file for this?

File: client/modules/accounts/templates/profile/profile.html [hunk 8]
```diff
@@ -35,12 +44,60 @@ <h3 class="panel-title" data-i18n="accountsUI.yourOrders">Your Orders</h3>
...
     </div>
-
+  {{#if isNotGuestOrAdmin }}
+    <!-- Upgrading button -->
+    <div class="flex-item">
+      <div class="panel panel-default">
+        <div class="panel-heading">
+          <h3 class="panel-title" >{{shopFormHeader}}</h3>
+          <i class="fa fa-chevron-down" aria-hidden="true" data-toggle="collapse" data-target="#showVendorForm" style="float:right;margin-top: -15px;"></i>
+					<span style="color:green; float:right;margin-top: -15px;margin-right:30px;">{{updatedShop}}</span>
```
Comment (andela-jare) [inline client/modules/accounts/templates/profile/profile.html:54]: Is there any reason for the in-line styles?

File: client/modules/accounts/templates/profile/profile.js [hunk 9]
```diff
@@ -64,5 +88,87 @@ Template.accountProfile.helpers({
       return "addressBookGrid";
     }
     return "addressBookAdd";
+  },
+  // to check if user is a vendor
+  isVendor: function () {
```
Comment (andela-jare) [inline client/modules/accounts/templates/profile/profile.js:93]: Why don't you use arrow function here?

**Pull request closed without merging**